src/main/java/com/team766/framework3/Procedure.java

```diff
@@ -1,48 +1,78 @@
 package com.team766.framework3;
 
 import com.google.common.collect.Sets;
-import com.team766.framework.LoggingBase;
+import com.team766.logging.Category;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+import java.util.Collection;
 import java.util.Set;
 
-public abstract class Procedure extends LoggingBase implements RunnableWithContext {
+public abstract class Procedure implements LoggingBase {
     // A reusable Procedure that does nothing.
-    private static final class NoOpProcedure extends Procedure {
+    private static final class NoOpProcedure extends InstantProcedure {
         @Override
-        public void run(final Context context) {}
+        public void run() {}
     }
 
-    public static final Procedure NO_OP = new NoOpProcedure();
+    public static final InstantProcedure NO_OP = new NoOpProcedure();
 
     private static int c_idCounter = 0;
 
     private static synchronized int createNewId() {
         return c_idCounter++;
     }
 
-    protected final int m_id;
-    private final Set<Mechanism<?>> reservations = Sets.newHashSet();
+    private final String name;
+    private final Set<Subsystem> reservations;
```

Comment:
You make a good point about the ownership checks, and it looks like the types mostly work out, so I've made the change to `Mechanism` everywhere.

It seems like WPILib interop is a recurring theme here, so I'd also be interested in having a conversation about that. I think we can assume a closed ecosystem around MF for this first version, but it would be good to have a strategy for this, on the theme of MaroonComponents.

There's also the longer-term aspiration of removing the need for `checkContextOwnership`, so that could also be a way to resolve this